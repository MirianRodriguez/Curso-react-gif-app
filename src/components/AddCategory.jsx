import { useState } from "react";
import PropTypes from "prop-types";

//para poder usar onNewCategory se debe desestructurar con llaves o hacer props.onNewCategory
export const AddCategory = ({onNewCategory}) => {
    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({target}) => {
        setInputValue(target.value);
    }

    const onSubmit = (event) => {
        event.preventDefault();
        if(inputValue.trim().length<=1) return;
        setInputValue('');
        onNewCategory(inputValue.trim());
    }

  return (
    <form onSubmit={onSubmit} aria-label="form">
        <input type="text" 
        placeholder="Buscar gif"
        value={inputValue}
        //llamada a la funciÃ³n onInputChange
        //no hace falta pasarle explicitamente el argumento event
        onChange={onInputChange}/>
    </form>
  )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}
