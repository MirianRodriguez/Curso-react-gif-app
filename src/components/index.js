//Archivo de barril: centraliza todas/varias las exportaciones 
//para que se puedan importar varios componentes desde el mismo archivo en una l√≠nea
export * from './AddCategory';
export * from './GifGrid';
