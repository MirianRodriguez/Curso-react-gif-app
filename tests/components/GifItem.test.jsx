import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem";

describe('Pruebas sobre el GifItem', () => {

    const title = 'un titulo';
    const url = 'https://una-url.jpg/';

    test('Debe coincidir con el snapshot', () => {
        const {container} = render(<GifItem title={title} url={url}/>);
        expect({container}).toMatchSnapshot();
    })

    test('Debe mostrar la img con src igual a la url enviada y alt igual al titulo', () => {
        render(<GifItem title={title} url={url}/>);
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    })

    test('Debe mostrarse el tÃ­tulo enviado', () => {
        render(<GifItem title={title} url={url}/>);
        expect(screen.getByText(title)).toBeTruthy();
    })
})