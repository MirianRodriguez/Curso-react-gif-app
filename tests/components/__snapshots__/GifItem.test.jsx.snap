// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el GifItem Debe coincidir con el snapshot 1`] = `
{
  "container": <div>
    <div
      class="card"
    >
      <img
        alt="un titulo"
        src="https://una-url.jpg/"
      />
      <p>
        un titulo
      </p>
    </div>
  </div>,
}
`;
